src/DeprecationDetector/DeprecationDetector.php

```diff
@@ -0,0 +1,85 @@
+<?php
+
+namespace SensioLabs\DeprecationDetector\DeprecationDetector;
+
+use SensioLabs\DeprecationDetector\AncestorResolver;
+use SensioLabs\DeprecationDetector\Finder\ParsedPhpFileFinder;
+use SensioLabs\DeprecationDetector\RuleSet\Loader\LoaderInterface;
+use SensioLabs\DeprecationDetector\Violation\Violation;
+use SensioLabs\DeprecationDetector\Violation\ViolationDetector;
+use SensioLabs\DeprecationDetector\Violation\Renderer\RendererInterface;
+
+class DeprecationDetector
+{
+    /**
+     * @var LoaderInterface
+     */
+    private $ruleSetLoader;
+
+    /**
+     * @var ParsedPhpFileFinder
+     */
+    private $deprecationFinder;
+
+    /**
+     * @var ViolationDetector
+     */
+    private $violationDetector;
+
+    /**
+     * @var RendererInterface
+     */
+    private $renderer;
+
+    /**
+     * @param LoaderInterface   $ruleSetLoader
+     * @param AncestorResolver  $ancestorResolver
+     * @param ParsedPhpFileFinder   $deprecationFinder
+     * @param ViolationDetector $violationDetector
+     * @param RendererInterface $renderer
+     */
+    public function __construct(
+        LoaderInterface $ruleSetLoader,
+        AncestorResolver $ancestorResolver,
+        ParsedPhpFileFinder $deprecationFinder,
+        ViolationDetector $violationDetector,
+        RendererInterface $renderer
+    ) {
+        $this->ruleSetLoader = $ruleSetLoader;
+        $this->ancestorResolver = $ancestorResolver;
+        $this->deprecationFinder = $deprecationFinder;
+        $this->violationDetector = $violationDetector;
+        $this->renderer = $renderer;
+    }
+
+    /**
+     * @param string $sourceArg
+     * @param string $ruleSetArg
+     *
+     * @return Violation[]
+     *
+     * @throws \Exception
+     */
+    public function checkForDeprecations($sourceArg, $ruleSetArg)
+    {
+        $ruleSet = $this->ruleSetLoader->loadRuleSet($ruleSetArg);
+
+        // TODO: not hard coded
+        $lib = (is_dir($ruleSetArg) ? $ruleSetArg : realpath('vendor'));
```

Comment:
The vendor directory could be read from the composer.json if it is defined in there, else I think it is ok to have a default hard coded